src/index.js

```diff
@@ -313,20 +333,27 @@ async function watch(stateManager, mainIndex) {
     listener.on('change', async change => {
       if (change.deleted === true) {
         await mainIndex.deleteObject(change.id);
-        log.info(`🛰 Deleted ${change.id}`);
+        log.info(`🛰  Deleted ${change.id}`);
       }
       changesConsumer.push(change, err => {
         if (err) {
           reject(err);
         }
       });
     });
-    listener.on('error', reject);
+
+    listener.on('error', err => {
+      sentry.report(err, {
+        retry: true,
+      });
+      log.info('🛰  Retrying watch', retry + 1);
+      watch(stateManager, mainIndex, retry + 1, err);
+    });
   });
 }
 
-async function error(err) {
-  sentry.report(err);
+async function error(err, extra) {
+  sentry.report(err, extra);
```

Comment:
This function is never called directly, it can not have extra.

Reply:
The reject should bubble all the way up to main, no?

Reply:
true, I didn't saw your second params on the reject. Not sure it work though `reject` only accept one param